packages/obojobo-express/models/media.js

```diff
@@ -0,0 +1,348 @@
+					INSERT INTO binaries
+						(blob, file_size, mime_type)
+					VALUES
+						($[binary], $[size], $[mimetype])
+					RETURNING *`,
+					{ binary, size, mimetype }
+				)
+				.then(result => {
+					mediaBinaryInfo = result
+
+					// If an orginal image is being stored, this query will create a record that the user uploaded an
+					// image. From this record, the user's original image and resized images derived from the original
+					// can be found. All resized images derived from an original will have the same mediaId as the
+					// original, so it is not necessary to store a reference to the resized images in the media table.
+					if (mode === MODE_INSERT_ORIGINAL_IMAGE) {
+						return transactionDb.one(
+							`
+							INSERT INTO media
+								(user_id, file_name)
+							VALUES
+								($[userId], $[filename])
+							RETURNING *`,
+							{ userId, filename }
+						)
+					}
+				})
+				.then(result => {
+					// If a resized image is being stored, a new media record was not created. The mediaId of the
+					// original image the resized image was derived from must be passed in and used.
+					const mediaRecordId = mode === MODE_INSERT_ORIGINAL_IMAGE ? result.id : mediaId
+
+					// The media_binaries table links media records to their respective binary information. This
+					// query creates that link. Both original and resized images must be linked. Since resized
+					// images have the same id as the original they are derived from, a dimensions field is
+					// stored to describe the binary the link is pointing to.
+					return transactionDb.one(
+						`
+							INSERT INTO media_binaries
+								(media_id, binary_id, dimensions)
+							VALUES
+								($[mediaRecordId], $[mediaBinariesId], $[dimensions])
+							RETURNING *`,
+						{ mediaRecordId, mediaBinariesId: mediaBinaryInfo.id, dimensions }
+					)
+				})
+		})
+	}
+
+	static cacheImageInDb(imageBinary, imageMetadata, imageDimensions, originalImageId) {
+		// sharp does not represent mimetypes as image/{mimetype}, transforming to this format keeps database consistent
+		const mimetype = `image/${imageMetadata.format}`
+		return (
+			Media.storeImageInDb({
+				binary: imageBinary,
+				size: imageMetadata.size,
+				mimetype,
+				dimensions: imageDimensions,
+				mode: MODE_INSERT_RESIZED_IMAGE,
+				mediaId: originalImageId,
+				userId: null
+			})
+				.then(newMediaRecord => {
+					return newMediaRecord.media_id
+				})
+				// catches any errors from transaction queries
+				.catch(err => {
+					logger.error(err)
+					throw err
+				})
+		)
+	}
+	static isValidFileType(filename, mimetype) {
+		const allowedFileTypes = new RegExp(mediaConfig.allowedMimeTypesRegex)
+
+		// test for valid mimetype
+		const isAllowerMimetype = allowedFileTypes.test(mimetype)
+		// test for valid extensions
+		const isAllowedExt = allowedFileTypes.test(path.extname(filename).toLowerCase())
+
+		if (!isAllowerMimetype || !isAllowedExt) {
+			return false
+		}
```

Comment:
Instead of the return false/return true you can just do `return isAllowedMimetype && isAllowedExt`